projects/knora-ui/src/lib/viewer/values/time-value/time-input/time-input.component.ts

```diff
@@ -0,0 +1,231 @@
+      return this.userInputToTimestamp(userInput);
+    }
+    return null;
+  }
+
+  set value(timestamp: string | null) {
+    if (timestamp !== null) {
+      const dateTime = this.convertTimestampToDateTime(timestamp);
+      this.form.setValue({date: dateTime[0], time: dateTime[1]});
+    } else {
+      this.form.setValue({date: null, time: null});
+    }
+    this.stateChanges.next();
+  }
+
+  @Input() errorStateMatcher: ErrorStateMatcher;
+
+  constructor(fb: FormBuilder,
+              @Optional() @Self() public ngControl: NgControl,
+              private fm: FocusMonitor,
+              private elRef: ElementRef<HTMLElement>,
+              @Optional() _parentForm: NgForm,
+              @Optional() _parentFormGroup: FormGroupDirective,
+              _defaultErrorStateMatcher: ErrorStateMatcher) { 
+
+    super(_defaultErrorStateMatcher, _parentForm, _parentFormGroup, ngControl);
+
+    this.dateFormControl = new FormControl(null);
+    this.dateFormControl.setValidators([Validators.required]);
+    
+    this.timeFormControl = new FormControl({value: null, Validators: [Validators.required, this.timeValidator]});
+
+    this.form = fb.group({
+      date: this.dateFormControl,
+      time: this.timeFormControl
+    });
+
+    fm.monitor(elRef.nativeElement, true).subscribe(origin => {
+      this.focused = !!origin;
+      this.stateChanges.next();
+    });
+
+    if (this.ngControl != null) {
+      this.ngControl.valueAccessor = this;
+    }
+  }
+
+  ngDoCheck() {
+    if (this.ngControl) {
+      this.updateErrorState();
+    }
+  }
+
+  ngOnDestroy() {
+    this.stateChanges.complete();
+  }
+
+  onContainerClick(event: MouseEvent) {
+    if ((event.target as Element).tagName.toLowerCase() != 'input') {
+      this.elRef.nativeElement.querySelector('input').focus();
+    }
+  }
+
+  writeValue(datetime: string | null): void {
+    this.value = datetime;
+  }
+
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+  }
+
+  _handleInput(): void {
+    this.onChange(this.value);
+  }
+
+  // return converted Date obj as a string without the milliseconds
+  userInputToTimestamp(userInput: any): string {
+    let splitTime = userInput.time.split(":");
+    const updateDate = new Date(userInput.date.calendarStart.year,
+                                (userInput.date.calendarStart.month - 1),
+                                userInput.date.calendarStart.day,
+                                splitTime[0],
+                                splitTime[1]
+    );
+
+    return updateDate.toISOString().split('.')[0]+"Z";
+  }
+
+  // converts and returns a unix timestamp string as an array consisting of a GregorianCalendarDate and a string
+  convertTimestampToDateTime(timestamp: string): { gcd: GregorianCalendarDate, time: string }[] {
```

Comment:
Could you make the return type an object with one member being time and the other the date?

What is `{ gcd: GregorianCalendarDate, time: string }[]` supposed to mean? An array of objects each having a member `gcd` and `time`?

Reply:
Yeah, returning an object would probably be better. Currently it's returning an array of objects because `gcd` and `time` are different types

Reply:
fixed in bfec41b